common/src/main/scala/azure/NotificationHubClient.scala

```diff
@@ -77,17 +77,29 @@ class NotificationHubClient(val notificationHubConnection: NotificationHubConnec
   }
 
   def sendNotification(rawPush: RawPush): Future[HubResult[Option[String]]] = {
-    val serviceBusTags = rawPush.tagQuery.map(tagQuery => "ServiceBusNotification-Tags" -> tagQuery).toList
-    logger.debug(s"Sending Raw Notification: $rawPush")
-    rawPush.post(
-      request(Endpoints.Messages)
-        .addHttpHeaders(rawPush.extraHeaders: _*)
-        .addHttpHeaders("ServiceBusNotification-Format" -> rawPush.format)
-        .addHttpHeaders(serviceBusTags: _*)
-    ).map {
-      case r if r.isSuccess => Right(r.header("Location"))
-      case r => Left(XmlParser.parseError(r))
+    def retry(count: Int): Future[HubResult[Option[String]]] = {
+      val serviceBusTags = rawPush.tagQuery.map(tagQuery => "ServiceBusNotification-Tags" -> tagQuery).toList
+      logger.debug(s"Sending Raw Notification: $rawPush")
+      rawPush.post(
+        request(Endpoints.Messages)
+          .addHttpHeaders(rawPush.extraHeaders: _*)
+          .addHttpHeaders("ServiceBusNotification-Format" -> rawPush.format)
+          .addHttpHeaders(serviceBusTags: _*)
+      ).flatMap {
+        case r if r.isSuccess =>
```

Comment:
I find `r` hard to read as in request/response environment it can mean either. Is `resp` better?